### PR TITLE
Refactor project filter in Project Manager

### PR DESCRIPTION
**Pull request by stijn-h**
- Reordered search bar and side bar buttons.
 - Removed ProjectListFilter (unnecessarily complicated abstraction).
 - Removed some unused variables/methods.

Naamloos

### CODE REVIEW

**Comment by YeldhamDev** [comment]
LGTM

**Review by YeldhamDev — APPROVED** [review]

**Review by Calinou — APPROVED** [review]
Nice work :slightly_smiling_face:

It feels like an upgrade over the existing design.

**Comment by akien-mga** [comment]
Thanks!

File: editor/project_manager.cpp [hunk 1]
```diff
@@ -1391,9 +1392,9 @@ void ProjectList::set_search_term(String p_search_term) {
 	_search_term = p_search_term;
 }
 
-void ProjectList::set_order_option(ProjectListFilter::FilterOption p_option) {
+void ProjectList::set_order_option(int p_option) {
 	if (_order_option != p_option) {
-		_order_option = p_option;
+		_order_option = (FilterOption)p_option;
```
[the remark this answers is not in the document]
Comment (hinlopen) [reply editor/project_manager.cpp:1397]: It was casted from int->FilterOption before this commit too, it just had multiple extra layers before the value ended up being passed to this function. Ultimately, the option button can only return the item index and so it must be casted to FilterOption. Your solution would involve redirecting the signal to an intermediate function that performs the cast and then calls set_order_option(), which is the same, but more complex.

**Pull request merged**